sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -36,11 +40,19 @@ case object Descending extends SortDirection {
   override def sql: String = "DESC"
 }
 
+case object NullFirst extends NullOrdering{
+  override def sql: String = "NULLS FIRST"
+}
+
+case object NullLast extends NullOrdering{
+  override def sql: String = "NULLS LAST"
+}
+
 /**
  * An expression that can be used to sort a tuple.  This class extends expression primarily so that
  * transformations over expression will descend into its child.
  */
-case class SortOrder(child: Expression, direction: SortDirection)
+case class SortOrder(child: Expression, direction: SortDirection, nullOrder: NullOrdering = null)
```

Comment:
Without giving default value for this parameter, there are about no more than 100 lines of code change in different places, including test cases. Setting "nulls first" as default behavior may break a lot of test cases that put nulls first when ASC, and last when DESC. The original default behavior will change. If this is expected, I will make the change. Thanks!